src/core/ui/forms/MarkdownInputControls/InsertEmbedCodeButton/InsertEmbedCodeButton.tsx

```diff
@@ -0,0 +1,207 @@
+import { BlockTitle } from '@/core/ui/typography';
+import { Actions } from '@/core/ui/actions/Actions';
+import DialogHeader from '@/core/ui/dialog/DialogHeader';
+import DialogWithGrid from '@/core/ui/dialog/DialogWithGrid';
+import { MARKDOWN_TEXT_LENGTH } from '@/core/ui/forms/field-length.constants';
+
+interface InsertEmbedCodeButtonProps extends IconButtonProps {
+  editor: Editor | null;
+  onDialogOpen?: () => void;
+  onDialogClose?: () => void;
+}
+
+const grey = {
+  50: '#F3F6F9',
+  200: '#DAE2ED',
+  300: '#C7D0DD',
+  700: '#434D5B',
+  900: '#1C2025',
+};
+const black = {
+  200: '#303030',
+  400: '#171717',
+  600: '#131313',
+};
+const validSources = [
+  'https://issuu.com', // Important - not tested because in order to get embed code from this site one must have account with paid plan!
+  'https://www.youtube.com',
+  'https://player.vimeo.com',
+  'https://demo.arcade.software',
+];
+
+export const InsertEmbedCodeButton = ({
+  editor,
+  onDialogOpen,
+  onDialogClose,
+  ...buttonProps
+}: InsertEmbedCodeButtonProps) => {
+  const [src, setSrc] = useState('');
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+
+  const { t } = useTranslation();
+
+  const notify = useNotification();
+
+  const buttonRef = useRef<HTMLButtonElement>(null);
+
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+
+  const handleOnCloseDialog = () => {
+    setIsDialogOpen(false);
+    editor?.commands.focus();
+    onDialogClose?.();
+    setSrc('');
+  };
+
+  const Textarea = styled(BaseTextareaAutosize)(
+    ({ theme }) => `
+    box-sizing: border-box;
+    width: 320px;
+    font-family: 'IBM Plex Sans', sans-serif;
+    font-size: 0.875rem;
+    font-weight: 400;
+    line-height: 1.5;
+    padding: 12px;
+    border-radius: 12px 12px 0;
+    color: ${theme.palette.mode === 'dark' ? grey[300] : grey[900]};
+    background: ${theme.palette.mode === 'dark' ? grey[900] : '#fff'};
+    border: 1px solid ${theme.palette.mode === 'dark' ? grey[700] : grey[200]};
+    box-shadow: 0 2px 2px ${theme.palette.mode === 'dark' ? grey[900] : grey[50]};
+
+    &:hover {
+      border-color: ${black[400]};
+    }
+
+    &:focus {
+      outline: 0;
+      border-color: ${black[400]};
+      box-shadow: 0 0 0 1px ${theme.palette.mode === 'dark' ? black[600] : black[200]};
+    }
+
+    /* firefox */
+    &:focus-visible {
+      outline: 0;
+    }
+  `
+  );
+
+  const handleOnIconButtonClick = () => {
+    setIsDialogOpen(true);
+  };
+
+  const handleOnTextareaChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
+    const cursorPosition = event.target.selectionStart;
+
+    setSrc(event.target.value);
+
+    setTimeout(() => {
+      if (textareaRef.current) {
+        textareaRef.current.selectionStart = cursorPosition;
+        textareaRef.current.selectionEnd = cursorPosition;
+      }
+    }, 0);
+  };
+
+  const handleOnSubmitIframe = () => {
+    const parser = new DOMParser();
```

Comment:
It's probably not in the requisites, but it would be very nice if we could just paste a youtube link instead of the entire iframe code

Reply:
This is not possible because the regular YouTube link from the user's browser address bar is different from the YouTube link for the iframe.